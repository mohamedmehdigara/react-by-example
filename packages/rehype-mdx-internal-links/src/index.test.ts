import mdxHastToJsx from '@mdx-js/mdx/mdx-hast-to-jsx'
import rehype from 'rehype'
import { Processor } from 'unified'
import rehypeMdxInternalLinks from '.'

const fixtures = {
  simple: `
    <a href="/is-relative" target="_blank" class="relative">
      relative link
    </a>

    <a href="#is-anchor" id="is-anchor" class="anchor">
      anchor link
    </a>

    <a href="https://github.com/absolute/link" target="_blank" class="absolute">
      absolute link
    </a>
  `,
}
describe('rehypeMdxInternalLinks', () => {
  let processor: Processor

  beforeEach(() => {
    processor = rehype()
      .use(rehypeMdxInternalLinks, {
        linkComponentShortcode: 'TheLinkComponent',
        linkComponentHrefProp: 'theHrefProp',
        filter: (el) => !String(el.properties?.href).startsWith('#'),
      })
      .use(() => ({ children }) => {
        return {
          type: 'element',
          tagName: '',
          children,
        }
      })
      .use(mdxHastToJsx)
  })

  it.each(Object.entries(fixtures))(
    'should rewrite links "%s"',
    async (_name, html) => {
      const result = await processor.process(html)
      expect(result.toString()).toMatchSnapshot()
    },
  )
})
