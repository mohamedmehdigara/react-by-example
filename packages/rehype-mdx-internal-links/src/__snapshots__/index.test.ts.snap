// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rehypeMdxInternalLinks should rewrite links "simple" 1`] = `
"<><html><head parentName=\\"html\\"></head><body parentName=\\"html\\"><TheLinkComponent parentName=\\"body\\" {...{\\"theHrefProp\\":\\"/is-relative\\",\\"target\\":\\"_blank\\",\\"className\\":\\"relative\\"}}>{\`
      relative link
    \`}</TheLinkComponent>{\`

    \`}<a parentName=\\"body\\" {...{\\"href\\":\\"#is-anchor\\",\\"id\\":\\"is-anchor\\",\\"className\\":\\"anchor\\"}}>{\`
      anchor link
    \`}</a>{\`

    \`}<a parentName=\\"body\\" {...{\\"href\\":\\"https://github.com/absolute/link\\",\\"target\\":\\"_blank\\",\\"className\\":\\"absolute\\"}}>{\`
      absolute link
    \`}</a>{\`
  \`}</body></html></>"
`;
